dotCMS/src/main/java/com/dotcms/rest/api/v1/page/NavResource.java

```diff
@@ -91,13 +93,17 @@ protected NavResource(final WebResource webResource) {
     @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
     @Path("/{uri: .*}")
     public final Response loadJson(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
-            @PathParam("uri") final String uri, @QueryParam("depth") final String depth) {
+            @PathParam("uri") final String uri, @QueryParam("depth") final String depth, @QueryParam("languageId") final String languageId) {
 
         final InitDataObject auth = webResource.init(false, request, true);
         final User user = auth.getUser();
 
         try {
-            final int maxDepth = Integer.parseInt(depth);
+            final int maxDepth = (UtilMethods.isSet(depth)) ? Integer.parseInt(depth) :
+                    defaultDepth;
+
+            final long langId = (UtilMethods.isSet(languageId)) ? Long.parseLong(languageId) :
```

Comment:
ConversionUtils.toLong(languageId, WebAPILocator.getLanguageWebAPI().getLanguage(request).getId());

If you want to do it more efficient, you can create a method such as

public static long toLong (final Object input, final Supplier<Long> defaultLong) {

and call it such as
ConversionUtils.toLong(languageId, ()->WebAPILocator.getLanguageWebAPI().getLanguage(request).getId());

it will avoid to call the Lang Api if it is not needed

Reply:
Same reason than above.